src/contracts/nft/nft/src/lib.rs

```diff
@@ -127,37 +172,50 @@ impl Contract {
     #[payable]
     pub fn mint_tweet_request(
         &mut self,
-        tweet_id: String,
-        image_url: String,
-        notify: String,
+        input_data: String,
+        signature: Vec<u8>,
     ) -> (AccountId, u64, String) {
+        let input: InputMintRequest = serde_json::from_str(&input_data).unwrap();
+        let signature = Signature::from_slice(&signature.as_slice());
+        require!(signature.is_ok(), "Invalid signature");
+        let signature = signature.unwrap();
+        let verifying_key = VerifyingKey::from_sec1_bytes(self.agent_verification_key.as_slice())
+            .ok()
+            .expect("verifying_key must be correct");
+
         require!(
-            env::attached_deposit().eq(&MIN_DEPOSIT),
+            env::attached_deposit()
+                .ge(&(MIN_DEPOSIT + self.compute_cost(input.public_metric.clone()))),
             format!(
                 "Minimum deposit Not met of {}, you attached {}",
-                &MIN_DEPOSIT,
+                &(MIN_DEPOSIT + self.compute_cost(input.public_metric.clone())),
                 env::attached_deposit()
             )
         );
-        if tweet_id.clone().parse::<u64>().is_err() {
+
+        assert!(verifying_key
```

Comment:
What's the purpose of signature verification?
The user that submits the transaction is signing said transaction it to even submit?
We're essentially verifying the signature of the principal submitter - no?

Reply:
Yes, the user signs what the bitte agent provides. the addition signature verification is to validate what was signed as a transaction came from agent. to ensure we got the right `Public_metrics` is correct and has not been updated

Reply:
The agent itself already query the tweet Data on mint intent, to check if the tweet_id is  valid and also uses the tweet description to  generate the AI image if needed. 
I simply used that data to compute the cost of the mint before the ZK-based process starts since the user pays upfront.